src/examples/pongxr/scene.js

```diff
@@ -0,0 +1,200 @@
+        ball.initPos = targetPosition;
+        scene.add(ball);
+        networking.remoteSync.addLocalObject(ball, { type: "ball", position: targetPosition }, true);
+    }
+}
+
+const initPlacement = () =>
+{
+    State.GameState = GameState.placement;
+    placementCube = PlacementCube();
+    scene.add(placementCube);
+    networking.remoteSync.addSharedObject(placementCube, 3, true);
+}
+
+/// GAME STATE
+
+State.eventHandler.addEventListener("gameover", (e) =>
+{
+    if (ball != undefined)
+    {
+        ball.reset();
+    } else
+    {
+        console.error("can't reset; no ball found!");
+    }
+});
+
+/// INPUT
+
+let doubleClick = false;
+
+State.eventHandler.addEventListener("select", e =>
+{
+    switch (State.GameState)
+    {
+        case (GameState.placement):
+            if (State.isMaster)
+            {
+                createPongLevel(placementCube);
+            }
+            break;
+
+        case (GameState.play):
+        default:
+            if (!doubleClick)
+            {
+                doubleClick = true;
+                setTimeout(function ()
+                {
+                    if (ball == undefined)
+                    {
+                        console.error("can't kickoff; no ball found!");
+                        return;
+                    }
+                    ball.kickoff();
+                    doubleClick = false;
+                }, 200);
+            }
+            else
+            {
+                if (State.debugMode) console.log("doubleclick");
+                // TOOD: implement dclick reset
+                // location.reload();
+            }
+            break;
+    }
+});
+
+// Add paddles when we know our inputs
+State.eventHandler.addEventListener("inputsourceschange", e =>
+{
+    if (paddles.length != 0) return //  avoid false positives, i.e headset put down, but paddles already instantiated
+
+    XRInput.controllerGrips.forEach((e, i) =>
+    {
+        const paddle = new Paddle();
+        scene.add(paddle);
+        paddles.push(paddle);
+        networking.remoteSync.addLocalObject(paddle, { type: "paddle" }, true);
+    });
+
+    // local paddle controller component to control player's networked paddle
+    // note: *all* XRInput data is local. 
+    // only the objects sync'd to it are networked.
+    // note2: we don't add it to rAF via Update() due to accumulated rAF lagginess over time.
+
+    setInterval(function ()
+    {
+        paddles.forEach((paddle, i) =>
+        {
+            if (XRInput.controllerGrips[ i ] != undefined)
+            {
+                paddle.position.lerp(XRInput.controllerGrips[ i ].position, .5);
+                paddle.quaternion.slerp(XRInput.controllerGrips[ i ].quaternion, .5);
+            }
+        });
+    }, 50);
+});
+
+/// NETWORKING 
+
+networking.remoteSync.addEventListener("open", (e) =>
+{
+    initPlacement();
+});
+
+networking.remoteSync.addEventListener('add', (destId, objectId, info) =>
```

Comment:
This is cool. Have you thought of any other ways to do this? My mind is wandering but I haven't thought of anything yet. Would love to brainstorm with you.

Reply:
ah yes, turns out this is how `three-network` [wants to be handled](https://github.com/takahirox/ThreeNetwork#concept). This whole pattern is weird imo - we end up adding each object twice, locally. Had to triple check but this is indeed [the prescribed way to do this](view-source:https://takahirox.github.io/ThreeNetworkDemo/firebase2.html?832) (also why I prefer shared objects whenever I get a chance; much simpler API, easier to follow). 
Let's definitely discuss!

Reply:
Out of scope. We'll follow up after this PR.